src/trace/ord/merge_batcher/mod.rs

```diff
@@ -196,42 +220,68 @@ impl<D: Ord, R: MonoidValue> MergeSorter<D, R> {
         while self.queue.len() >= 2 {
             let list1 = self.queue.pop().unwrap();
             let list2 = self.queue.pop().unwrap();
+
             let merged = self.merge_by(list1, list2);
-            self.queue.push(merged);
+            if !merged.is_empty() {
+                self.queue.push(merged);
+            }
         }
 
         if let Some(mut last) = self.queue.pop() {
-            swap(&mut last, target);
+            // TODO: Reuse the `last` buffer somehow
+            if target.capacity() == 0 {
```

Comment:
Why is this implementation better that the original?

Reply:
It takes advantage of any capacity the target vec has

Reply:
Eh nvm, I'll remove it